tests/test_on_data/test_format_converters/test_spikeglx_converter.py

```diff
@@ -30,43 +29,45 @@ def setUp(self):
     def tearDown(self):
         rmtree(self.tmpdir)
 
-    def assertNWBFileStructure(self, nwbfile_path: FilePath):
+    def assertNWBFileStructure(self, nwbfile_path: FilePath, expected_session_start_time: datetime):
         with NWBHDF5IO(path=nwbfile_path) as io:
             nwbfile = io.read()
 
-            expected_session_start_time = datetime(2020, 11, 3, 10, 35, 10).astimezone()
             assert nwbfile.session_start_time == expected_session_start_time
 
-            assert "ElectricalSeriesAP" in nwbfile.acquisition
-            assert "ElectricalSeriesLF" in nwbfile.acquisition
+            assert "ElectricalSeriesAPImec0" in nwbfile.acquisition
+            assert "ElectricalSeriesLFImec0" in nwbfile.acquisition
             assert "ElectricalSeriesNIDQ" in nwbfile.acquisition
+            assert len(nwbfile.acquisition) == 3
 
-            assert "Neuropixel-Imec" in nwbfile.devices
+            assert "NeuropixelImec0" in nwbfile.devices
+            assert "NIDQBoard" in nwbfile.devices
+            assert len(nwbfile.devices) == 2
 
             assert "NIDQChannelGroup" in nwbfile.electrode_groups
-            assert "s0" in nwbfile.electrode_groups
+            assert "Imec0Shank0" in nwbfile.electrode_groups
+            assert len(nwbfile.electrode_groups) == 2
 
     def test_single_probe_spikeglx_converter(self):
         converter = SpikeGLXConverterPipe(folder_path=SPIKEGLX_PATH / "Noise4Sam_g0")
-        # import json
         metadata = converter.get_metadata()
 
         test_metadata = deepcopy(metadata)
         for exclude_field in ["session_start_time", "identifier"]:
             test_metadata["NWBFile"].pop(exclude_field)
-        expected_metadata = load_dict_from_file(file_path=Path(__file__).parent / "single_probe_metadata.json")
-        neuroconv_version = importlib.metadata.version("neuroconv")
+        expected_metadata = load_dict_from_file(file_path=Path(__file__).parent / "spikeglx_single_probe_metadata.json")
 
         # Exclude watermarks from testing assertions
         del test_metadata["NWBFile"]["source_script"]
         del test_metadata["NWBFile"]["source_script_file_name"]
 
-        self.assertDictEqual(d1=test_metadata, d2=expected_metadata)
+        assert test_metadata == expected_metadata
```

Comment:
Updating this since the better diff helped with debugging

Reply:
Great.